tests/trainer/test_trainer_eval.py

```diff
@@ -52,6 +52,28 @@ def test_eval_call():
     assert trainer.state.eval_metrics['eval']['MulticlassAccuracy'].compute() != 0.0
 
 
+@world_size(1, 2)
+@pytest.mark.parametrize('size', [12, 13, 14, 15, 16])
+@pytest.mark.parametrize('batch_size', [1, 2, 3, 4, 6])
+@pytest.mark.filterwarnings(r'ignore:Cannot split tensor of length.*:UserWarning')
+def test_eval_with_nondivisible_dataset(world_size, size, batch_size):
+    # Construct the trainer
+    trainer = Trainer(model=ZeroModel())
+
+    # Evaluate the model
+    dataset = ParityDataset(size=size)
+    trainer.eval(eval_dataloader=DataLoader(
+        dataset=dataset,
+        batch_size=batch_size,
+        sampler=dist.get_sampler(dataset),
+    ))
+
+    expected_acc = 1 - (size // 2) / size
+    metric = trainer.state.eval_metrics['eval']['MulticlassAccuracy']
+    assert metric.compute() - expected_acc < 1e-5
+    assert metric.tp + metric.tn + metric.fp + metric.fn == size  # type: ignore
```

Comment:
```suggestion
    assert metric.tp.item() + metric.fn.item()  == size  # type: ignore
```
the way torchmetrics handles `MulticlassAccuracy`, adding everything together is double counting. see https://github.com/Lightning-AI/torchmetrics/blob/17c0e9fdf636cff7299fc3d86fa5e3709b603737/src/torchmetrics/functional/classification/accuracy.py#L76-L83